samples/iolib/src/main/cpp/player/OneShotSampleSource.cpp

```diff
@@ -22,20 +22,32 @@
 
 namespace iolib {
 
-void OneShotSampleSource::mixAudio(float* outBuff, int32_t numFrames) {
+void OneShotSampleSource::mixAudio(float* outBuff, int numChannels, int32_t numFrames) {
     int32_t numSampleFrames = mSampleBuffer->getNumSampleFrames();
     int32_t numWriteFrames = mIsPlaying
                          ? std::min(numFrames, numSampleFrames - mCurFrameIndex)
                          : 0;
 
     if (numWriteFrames != 0) {
         // Mix in the samples
-        int32_t lastIndex = mCurFrameIndex + numWriteFrames;
+        // int32_t lastIndex = mCurFrameIndex + numWriteFrames;
 
         // investigate unrolling this loop...
         const float* data  = mSampleBuffer->getSampleData();
-        for(int32_t index = 0; index < numWriteFrames; index++) {
-            outBuff[index] += data[mCurFrameIndex++];
+        if (numChannels == 1) {
+            // MONO output
+            for (int32_t frameIndex = 0; frameIndex < numWriteFrames; frameIndex++) {
+                outBuff[frameIndex] += data[mCurFrameIndex++];
+            }
+        } else if (numChannels == 2) {
+            // STEREO output
+            int dstSampleIndex = 0;
+            float leftGain = -(-mPan + 1.0) / 2.0f;
```

Comment:
Why is leftGain negative? That could cause phase cancellation when centered.
How about:

    float rightGain = (mPan * 0.5) + 0.5;
    float leftGain = 1.0 - leftGain;

Also check out "Pan Laws" on this page:
http://www.cs.cmu.edu/~music/icm-online/readings/panlaws/

Reply:
Clumsy arithmetic on my part. Done.